src/utils/api.js

```diff
@@ -0,0 +1,68 @@
+const API_BASE_URL =
+  process.env.NODE_ENV === "production"
+    ? "https://api.articlelist.ignorelist.com"
+    : "http://localhost:3001";
+
+// Fetch articles by a user
+export const getArticlesByUser = async (token) => {
+  try {
+    const response = await fetch(`${API_BASE_URL}/articles`, {
+      method: "GET",
+      headers: { Authorization: `Bearer ${token}` },
+    });
+    if (!response.ok) {
+      throw new Error("Network response was not ok");
+    }
+    const articles = await response.json();
```

Comment:
```js
    if (!response.ok) {
      throw new Error("Network response was not ok");
    }
    return await response.json();

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `checkResponse` that will check any response from the server if it’s ok or not.


```js
function checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(checkResponse)

```